h2o-admissibleml/src/main/java/hex/Infogram/InfogramUtils.java

```diff
@@ -0,0 +1,239 @@
+package hex.Infogram;
+
+import hex.Model;
+import hex.ModelBuilder;
+import hex.SplitFrame;
+import hex.deeplearning.DeepLearningModel;
+import hex.glm.GLMModel;
+import hex.schemas.DRFV3;
+import hex.schemas.DeepLearningV3;
+import hex.schemas.GBMV3;
+import hex.schemas.GLMV3;
+import hex.tree.drf.DRFModel;
+import hex.tree.gbm.GBMModel;
+import hex.tree.xgboost.XGBoostModel;
+import water.DKV;
+import water.Key;
+import water.Scope;
+import water.api.schemas3.ModelParametersSchemaV3;
+import water.fvec.Frame;
+import water.util.TwoDimTable;
+import static hex.Infogram.InfogramModel.InfogramParameters;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.DoubleStream;
+
+public class InfogramUtils {
+
+  /**
+   * This method will take the columns of _parms.train().  It will then remove the response, any columns in 
+   * _parms._sensitive_attributes from the columns of _parms.train(), weights_column, offset_column.  Then, the 
+   * columns that are left are the columns that are eligible to get their InfoGram.
+   *
+   * @param parms
+   * @return
+   */
+  public static String[] extractPredictors(InfogramParameters parms) {
```

Comment:
it looks like most (if not all) the logic in this method replicates what is done in `ModelBuilder.init` when preparing the `_train` frame, and therefore I'm afraid it could easily get out of sync if changes are done to `ModelBuilder`.
Is there any reason why you can't simply use the existing logic?

Reply:
You are right.  I use _train where ModelBuilder.init has done its magic.  Just need to remove the repsone_column and protected_columns